modules/gapi/src/compiler/gislandmodel.cpp

```diff
@@ -335,6 +335,24 @@ ade::NodeHandle GIslandModel::producerOf(const ConstGraph &g, ade::NodeHandle &d
     return ade::NodeHandle();
 }
 
+std::string GIslandModel::buildIslandName(const ade::NodeHandle& island_nh, const Graph& g) {
+    auto island_ptr = g.metadata(island_nh).get<FusedIsland>().object;
+    std::string island_name = island_ptr->name();
+
+    std::string backend_name = "";
+    std::string backend_impl_type_name = typeid(island_ptr->backend().priv()).name();
+    std::regex rgx("G([a-zA-Z0-9]+)BackendImpl");
+    std::smatch matches;
+    if (std::regex_search(backend_impl_type_name, matches, rgx)) {
+        if (2ul == matches.size()) {
+            backend_name = matches[1].str();
+            island_name = island_name + "_" + backend_name;
+        }
+    }
```

Comment:
If match fails, I'd use the full backend type name instead of nothing.

Reply:
Full backend name will contains some mangled symbhols and numbers which are maps to "::" namespace resolution operator, is it applicable?

Reply:
Yes, please use the full name in that case.

Reply:
Thanks, fixed!